docker-compose.traefik.yml

```diff
@@ -0,0 +1,39 @@
+version: "3.3"
+
```

Comment:
Is there a reason why a new file was created instead of adding this to `docker-compose.yml`?

Reply:
Hi, i think he separated the files to show the difference between using traefik or not using it.

Reply:
While that may be the case, that's not a valid reason to ultimately have two files for people to use. How would they handle this? However, please let the PR author comment. I don't want us pointlessly speculating and confusing ourselves.

Reply:
I forgot to update the initial PR description so believe this confused you (I have updated it now). The idea behind these new stack files is as below:

* `docker-compose.yml` is for running a single wiki instance which is pre-configured to work with Traefik as a front-end
* `docker-compose.traefik.yml` is for running pre-configured Traefik instance that can serve one or more wiki instances

The reason of having it on separate files is that one could want to have multiple wikis running on a single VPS, having both Traefik and Wiki on a the same stack file would make this impossible without the file modification, but having it separated allows to easily spin necessary number of Wiki stacks and then run a single Traefik instance to serve all the Wikis

Please also see the README updates

Reply:
Thank you very much @vedmaka for the clarification, I'll read this and think about it some more.

Reply:
Okay, I see what this is proposing. Basically, we could have multiple separate Canasta containers running on the same server, but there should only be one Traefik instance serving all of them, routing each one to the correct path.

My concern behind this is it makes using this stack a whole lot more complicated. Right now, it's as simple as running `docker compose up -d` even if you don't have the Canasta CLI.

I feel like it would be alleviated if Canasta adds support for running a single-source wiki farm, as detailed in https://github.com/CanastaWiki/Canasta/issues/57 with support from members of the community. This way, we don't have to deal with a one-to-many mapping of a Traefik instance interfacing with multiple Canasta stacks.

Reply:
At the same time, I understand your use case, where people still need to run multiple Canasta stacks on their machine. I still am very hesitant about adding this type of complexity to Canasta, even though I understand why it's being done this way. We also can't get this merged in until the Canasta CLI adapts to offer support for this.

Reply:
Thanks @jeffw16, you did understand this right and your point on that this increases complexity compared to a simple `docker-compose up -d` is absolutely legit too.

I don't have a good solution for this though, since even running a single instance of wiki paired with a single Traefik requires an extra step to be performed by end user ( to create a docker network [[1]](https://github.com/CanastaWiki/Canasta-DockerCompose/pull/26/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R22) )

We probably could have both wiki and Traefik sitting on the same `docker-compose.yml` by default, this would mean that running a single wiki instance will stay as simple as possible ( though the initial network creation may still be necessary ) and if one would want to have multiple instances served at time - the README will contain information on how to split the Compose file and what commands to run

The alternative could be to remove the need of running docker commands by end user and instead have own entrypoint for the stack. We did something similar to that on https://github.com/WikiTeq/docker-wikiteq-stack ( see `compose.sh` ) when we wanted to split stack yml into separate pieces and eventually allow this to evolve into `config-available` / `config-enabled` pattern

It looks like the https://github.com/CanastaWiki/Canasta-CLI is an evolution for that, so maybe we could delegate all of the start configurations to the CLI and impose a rule that the stack has to be managed via CLI by end users? In this case the simplicity of `docker-compose up -d` won't be lost because the CLI can do all the necessary operations on `start` or similar command

In the end, since there is no silver bullet, I think it's up to you to choose which approach looks better

Reply:
Yeah, we are stuck in a tough position here. But I think the reason the proposal here to split the Traefik container to a separate YAML is because it can serve multiple Canasta container instances. Our 2023 GSoC project is to add wiki farm support. Let's revisit this PR as we get closer to the GSoC contributor finishing up their wiki farm project.

Reply:
We're well into the summer. How do we proceed here?

Reply:
Now that wiki farm support is released, is this PR still relevant? We no longer wish to support running multiple Canasta instances on the same server.

Reply:
The farm feature and multi-instance deployment are separate approaches with different goals and side effects, and it would be nice to be able to run it via Traefik as multi-container setup. Either way, it seems like this PR deserves a rework to move the Traefik setup into completely optional vs how it's done right now. It's also likely the Traefik Compose should not be part of it and instead be a separate stack